### DISCUSSION TITLE
JSON Manager crossed request issue?

### DISCUSSION THREAD

**Discussion by D8H** [4ian/GDevelop#3257] [Archive] [answered]
Shouldn't the callbacks be stacked when a file is being read?

https://github.com/4ian/GDevelop/blob/bc606ed1be28275624678dd7c6bcc192c2da846d/GDJS/Runtime/jsonmanager.ts#L111-L115

Because when `loadJson` is called for the same file at 2 places. This could happens:
* Alice asks the json
* Bob asks the json
* Alice's parsed json is added to `_loadedJsons`
* Alice gets call-backed
* Bob's parsed json is added to `_loadedJsons` over Alice one
* Bob gets call-backed

I guess that when there is already a callback for a given file, the second request should only adds the callback to the list instead of loading the file again.

**Comment by 4ian** [comment, marked as answer]
You're correct, a pending list (with proper cleaning if the loading failed I guess) is needed to have a 100% correct behavior including the case you're mentioning.
